### PR TITLE
Cleaner should finally be PR-able now

### PR DESCRIPTION
**Pull request by Nikozzh**
恭喜发财

### CODE REVIEW

**Comment by nicki419** [comment]
Did rebase from main.

**Comment by nicki419** [comment]
Multiple things wrong here. Will need to be fixed tomorrow.

1. Error when accessing Cleaner Page: `InvalidOperationException: Cannot provide a value for property 'TableController' on type 'Famicom.Components.Pages.Cleaner'. There is no registered service of type 'TableController.ITableController'.`
2. On the CleanerModel, you use Constructor injection to request a TableController. You then use that TableController to get all tables from the API. That is **not** how we should do it. The tables on the API might not necessarily be used. The algorithm on how to do it properly is simple: Retrieve all tables from the Database, then loop through them, using the TableControllerService to get the reference to the correct TableController (Which is in the TableControllerService, hence the adding it as an injectable service in Famicom's Program.cs) and use that TableController to a) get the status of the table, see if it's operational and b) set the height of that table accordingly.

File: Famicom/Models/CleanerModel.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                    return;
+                }
+
+                int targetHeight = isMaxHeight ? 1320 : 1000;
+
+                foreach (var guid in tableGuids)
+                {
+                    Console.WriteLine($"Setting table {guid} to height {targetHeight}mm.");
+
+                    try
```
[the remark this answers is not in the document]
Comment (Nikozzh) [reply Famicom/Models/CleanerModel.cs:36]: oki

File: Famicom/Components/Pages/Cleaner.razor.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    public partial class CleanerBase : ComponentBase
+    {
+        public bool IsCleaningMode { get; private set; }
+
+        private CleanerModel? cleanerModel;
+
+        protected CleanerModel? CleanerModel { get; set; }
+
+        [Inject]
+        public ITableController? TableController { get; set; }
```
[the remark this answers is not in the document]
Comment (Nikozzh) [reply Famicom/Components/Pages/Cleaner.razor.cs:19]: oki

File: Famicom/Components/Pages/Cleaner.razor.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+        protected override void OnInitialized()
+        {
+            if (TableController == null)
```
[the remark this answers is not in the document]
Comment (Nikozzh) [reply Famicom/Components/Pages/Cleaner.razor.cs:23]: oki

File: Famicom/Models/CleanerModel.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                    {
+                        await tableController.SetTableHeight(targetHeight, guid, new Progress<ITableStatusReport>());
+                        Console.WriteLine($"Successfully updated table {guid}.");
+                    }
+                    catch (Exception ex)
+                    {
+                        Console.WriteLine($"Failed to update table {guid}: {ex.Message}");
```
[the remark this answers is not in the document]
Comment (Nikozzh) [reply Famicom/Models/CleanerModel.cs:43]: oki

File: Famicom/Components/Pages/Cleaner.razor [hunk 5]
```diff
@@ -0,0 +1,21 @@
+﻿@page "/cleaner"
+@inherits CleanerBase
+
+<PageTitle>Cleaner Panel</PageTitle>
+
+<MudContainer MaxWidth="MaxWidth.Medium">
```
[the remark this answers is not in the document]
Comment (Nikozzh) [reply Famicom/Components/Pages/Cleaner.razor:6]: This is hard qwq

**Pull request closed without merging**